### PR TITLE
Add in-toto GSoC Project Idea

### PR DESCRIPTION
**Pull request by PradyumnaKrishna**
cc @adityasaky @SantiagoTorres @lukpueh

### CODE REVIEW

**Comment by nate-double-u** [comment]
Hi @PradyumnaKrishna, thanks for this. This project idea is late (we had hoped that we'd get project ideas in by March 18), but I think we could probably accept this -- you'll just get fewer eyes on the project idea, and fewer candidates writing proposals.

We still need maintainer approval to merge this in tho. @adityasaky @SantiagoTorres @lukpueh, could one of you approve (or thumbs up) this PR, preferably @SantiagoTorres since you're listed as a mentor on each project idea.

**Comment by nate-double-u** [comment]
If the maintainers don't want this for GSoC, these would both be good project ideas for [LFX Mentorship 2024 Term 2](https://github.com/cncf/mentoring/tree/main/programs/lfx-mentorship/2024/02-Jun-Aug). Term 2 is opening for project ideas in early April.

**Review by adityasaky — APPROVED** [review]

**Review by nate-double-u — APPROVED** [review]

**Pull request merged**
